airflow/contrib/operators/livy_operator.py

```diff
@@ -0,0 +1,178 @@
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+This module contains the Apache Livy operator.
+"""
+
+from time import sleep
+
+from airflow.contrib.hooks.livy_hook import BatchState, LivyHook
+from airflow.exceptions import AirflowException
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+
+
+class LivyOperator(BaseOperator):
+    """
+    This operator wraps the Apache Livy batch REST API, allowing to submit a Spark
+    application to the underlying cluster.
+
+    :param file: path of the file containing the application to execute (required).
+    :type file: str
+    :param class_name: name of the application Java/Spark main class.
+    :type class_name: str
+    :param args: application command line arguments.
+    :type args: list
+    :param jars: jars to be used in this sessions.
+    :type jars: list
+    :param py_files: python files to be used in this session.
+    :type py_files: list
+    :param files: files to be used in this session.
+    :type files: list
+    :param driver_memory: amount of memory to use for the driver process.
+    :type driver_memory: str
+    :param driver_cores: number of cores to use for the driver process.
+    :type driver_cores: str
+    :param executor_memory: amount of memory to use per executor process.
+    :type executor_memory: str
+    :param executor_cores: number of cores to use for each executor.
+    :type executor_cores: str
+    :param num_executors: number of executors to launch for this session.
+    :type num_executors: str
+    :param archives: archives to be used in this session.
+    :type archives: list
+    :param queue: name of the YARN queue to which the application is submitted.
+    :type queue: str
+    :param name: name of this session.
+    :type name: str
+    :param conf: Spark configuration properties.
+    :type conf: dict
+    :param proxy_user: user to impersonate when running the job.
+    :type proxy_user: str
+    :param livy_conn_id: reference to a pre-defined Livy Connection.
+    :type livy_conn_id: str
+    :param polling_interval: time in seconds between polling for job completion. Don't poll for values >=0
+    :type polling_interval: int
+    :param timeout: when greater than zero, seconds to wait for a terminal state before killing the batch.
+    :type timeout: int
+    """
+
```

Comment:
```suggestion

    template_fields = ('spark_params')

```

This would then let any of the params used to make up this dict be templated.